fedmsg_meta_fedora_infrastructure/fasshim.py

```diff
@@ -133,13 +131,20 @@ def _add_to_cache(users):
     global _fas_cache
 
     for user in users:
-        nicks = user.get('ircnicks', [])
-        for nick in nicks:
+        nick = user.get('ircnick')
+        if nick:
             _fas_cache[nick] = user['username']
-        
-        emails = user.get('emails', [])
-        for email in emails:
+        else:
+            nicks = user.get('ircnicks', [])
+            for nick in nicks:
+                _fas_cache[nick] = user['username']
+        email = user.get('email')
```

Comment:
Same here I don't think the user model has an `email` property, it's all in `emails`.

Reply:
Same as nick above.

Reply:
Ah I see. There's two ways of going at it, you can either deal with the complexity of the 2 sources here, or you can modify the results of the source that is going to be deprecated (FAS) to make it compatible with the new one (adding `ircnicks` and `emails` values).
I tend to favor the latter option because then when we remove the code of the deprecated option, we just delete the block and there's no specific handling anywhere else in the codebase that we must remember to remove too.

Reply:
ok, so really the way it was _before_ I created the add_to_cache method was better because it reduces coupling. If I revert that section I don't need to make anything compatible and we can just remove the fas sections once it is deprecated.